Graphs/TreeGraph/index.js

```diff
@@ -336,40 +184,61 @@ class TreeGraph extends AbstractGraph {
 
         // Update the nodes...
         const node = svg.selectAll('g.node')
-            .data(nodes, (d) => { return d.id || (d.id = ++i); });
+            .data(nodes, (d) => {
+                return d.id || (d.id = ++i);
+            });
 
         // Enter any new modes at the parent's previous position.
         const nodeEnter = node.enter().append('g')
             .attr('class', 'node')
             .attr("transform", (d) => {
-                return d.parent 
-                  ? "translate(" + d.parent.y + "," + d.parent.x + ")" 
-                  : "translate(" + source.y0 + "," + source.x0 + ")";
+                return d.parent ?
+                    "translate(" + d.parent.y + "," + d.parent.x + ")" :
+                    "translate(" + source.y0 + "," + source.x0 + ")";
             })
             .on('click', this.click);
 
-        // Add Circle for the nodes
-        nodeEnter.append('circle')
-            .attr('class', 'node')
-            .attr('r', 1e-6)
-            .style("fill", (d) => {
-                return d._children ? "lightsteelblue" : "#fff";
-            });
-
-        // Add labels for the nodes
-        nodeEnter.append('text')
-            .attr("dy", ".35em")
-            .attr("x", (d) => {
-                return d.children || d._children ? -13 : 13;
+        nodeEnter.append('g').append('rect')
+            .attr('rx', 3)
+            .attr('ry', 3)
+            .attr('width', rectNode.width)
+            .attr('height', rectNode.height)
+            .attr('class', 'node-rect')
+
+        nodeEnter.append('foreignObject')
+            .attr('x', rectNode.textMargin)
+            .attr('y', rectNode.textMargin)
+            .attr('width', () => {
+                return (rectNode.width - rectNode.textMargin * 2) < 0 ? 0 :
+                    (rectNode.width - rectNode.textMargin * 2)
             })
-            .attr("text-anchor", (d) => {
-                return d.children || d._children ? "end" : "start";
+            .attr('height', () => {
+                return (rectNode.height - rectNode.textMargin * 2) < 0 ? 0 :
+                    (rectNode.height - rectNode.textMargin * 2)
+            })
+            .append('xhtml').html((d) => {
+                let img = this.fetchImage(d.data.apiData, d.data.contextName);
```

Comment:
we need to specify the attribute names to display through config! Similarly how we specify which columns we want to show on a table. Not all object have name and description for example for a subnet you want to display the subnet in a CIDR format.